osu.Game/Skinning/Skin.cs

```diff
@@ -10,6 +10,8 @@ public abstract class Skin
     {
         public readonly SkinInfo SkinInfo;
 
+        public virtual SkinConfiguration Configuration { get; protected set; }

```

Comment:
So how do you envision mania skinning to work? And catch skinning? Is the endgame to add the CatcherDashColour and individual configurations for mania playstyles at a base `SkinConfiguration` level? What if we want to extend each ruleset with additional skin configurations in the future?

Reply:
Probably derived `SkinConfiguration` types, or maybe a level below that.